main.js

```diff
@@ -89,29 +89,35 @@ function getAll(){
 // main ranking function,
 function rank(players, games){
     function adjust(ps, game){
-        var winners = game.winner === 'l'? game.playersL : game.playersR
-        var losers = game.winner === 'r'? game.playersL : game.playersR
-        var allParticipants = [...winners, ...losers]
-        var winners = ps
-            .filter(p=>winners.includes(p.id))
-            .map(p=>({rank: 1, id:p.id, skill:p.skill}))
-        var losers = ps
-            .filter(p=>losers.includes(p.id))
-            .map(p=>({rank: 2, id:p.id, skill:p.skill}))
-        var toAdjust = [...winners, ...losers]
-        trueskill.AdjustPlayers(toAdjust)
-        return ps.map(p=>({
-            id: p.id,
-            skill: allParticipants.includes(p.id)?
-                toAdjust.filter(o=>o.id===p.id)[0].skill : p.skill
-        }))
+        var rankL, rankR;
+        if (game.winner === 'l') {
+            rankL = 1;
+            rankR = 2;
+        }
+        else if (game.winner === 'r') {
+            rankL = 2;
+            rankR = 1;
+        }
+        else {
+            console.log("Unsupported game winner ", game.winner);
+        }
+
+        var teamL = ps.filter(p=>game.playersL.includes(p.id));
+        var teamR = ps.filter(p=>game.playersR.includes(p.id));
+        var otherPlayers = ps.filter(
+            p=>!game.playersL.includes(p.id) && !game.playersR.includes(p.id));
+
+        var newRatings = calculateNewRatings(
+            GameInfo, teamL, teamR, rankL, rankR);
```

Comment:
I think `trueskill.js` should have at least a fake module style export, as in it should just end with `trueSkill2 = {calculateNewRatings, GameInfo}` and then here do `trueSkill2.calculateNewRatings(...)`

Reply:
Also, given we will never have the concept of a draw, would it be better for `calculateNewRatings` to just take `winners` and `losers`. (`GameInfo` could also remain private to `trueSkill2`).

Reply:
Maybe, but that just pushes calculating winners and losers to `main.js`. It makes sense to keep it here, especially if we want to extend to include the concept of a draw.

Reply:
Martyn, we're never gonna have a draw bro